### DISCUSSION TITLE
Automatic merging of topics that overlap in the embedding space

### DISCUSSION THREAD

**Discussion by serenalotreck** [MaartenGr/BERTopic#1729] [Q&A] [answered]
Thanks for such a fantastic tool!

I trained a topic model, and visualized the output, which looks like this:

![image](https://github.com/MaartenGr/BERTopic/assets/41377532/65e35116-aff6-4174-81cb-e9433a8b3127)

There are plenty of places where there are multiple topics directly on top of one another, and looking at the topic representations, it makes total sense to combine them into one mega topic. However, there are a lot of topics, so manually creating a list of which ones should be merged is difficult. But when I tried using `topic_model.reduce_topics(docs, nr_topics='auto')`, the output was the following:

![image](https://github.com/MaartenGr/BERTopic/assets/41377532/afd7238d-ad0c-4b29-9f4f-648e38b0e5c5)

Which totally destroys the sensical topics that came out in the first place -- I went ahead and made a list of the meta-topics I expected to see (based on the topics of the overlapping groups), and basically none of them came out after merging; there were just a couple of very generic topics.

Is there a better way to try merging, or will I have to mouse over the topic visualization and manually list the topic numbers that I want to merge?

**Comment by MaartenGr** [comment, marked as answer]
If you want to merge topics, I would advise following the [hierarchical topic modeling tutorial](https://maartengr.github.io/BERTopic/getting_started/hierarchicaltopics/hierarchicaltopics.html) instead. It gives a bit more freedom and control over merging.
